tools/myst/enc/syscall.c

```diff
@@ -819,10 +822,12 @@ static long _poll(struct pollfd* fds, nfds_t nfds, int timeout)
         goto done;
     }
 
-    /* copy back the revents field */
+    /* copy back the revents field and sanitize the value provided by the
+     * host-side to be a subset of the events mask.
+     */
     for (nfds_t i = 0; i < nfds; i++)
     {
-        fds[i].revents = copy[i].revents;
+        fds[i].revents = copy[i].revents & fds[i].events;
```

Comment:
I missed this before: according the the man page "The  bits  returned  in revents can include any of those specified in events, **or one of the values POLLERR, POLLHUP,  or POLLNVAL**.". So maybe:

```
const short int mask = POLLERR | POLLHUP | POLLNVAL;
fds[i].revents = copy[i].revents & (fds[i].events | mask);
```